src/writing-your-own-application/configuration.md

```diff
@@ -1 +1,25 @@
 # Configuration
+
+The [esp-config] crate provides a way how to manage configuration settings.
+
+
+## Usage
+
+The full list of available options could be found in `out` dir with the format `{prefix}_config_table.md`, for example `esp_hal_config_table.md` when changed options are `esp-hal`-related.
```

Comment:
IMHO that's a bit too much implementation details - we should mention they can find config options in the docs. Sending users to `./target/.../out` folders sounds somewhat scary